### DISCUSSION TITLE
Add support for SSLKEYLOGFILE

### DISCUSSION THREAD

**Discussion by studersi** [OpenIDC/mod_auth_openidc#813] [Ideas]
In version Apache 2.4.49, support for the SSLKEYLOGFILE environment variable was introduced to mod_ssl: https://dlcdn.apache.org/httpd/CHANGES_2.4
>  mod_ssl: Support logging private key material for use with
     wireshark via log file given by SSLKEYLOGFILE environment
     variable.  Requires OpenSSL 1.1.1.  PR 63391.  [Joe Orton]

The SSLKEYLOGFILE environment variable appears to turn into the de-facto standard for having the pre-master secrets logged in order to decrypt TLS traffic for debugging.

As far as I can tell, requests performed by mod_auth_openidc do not support logging the pre-master secrets by setting the SSLKEYLOGFILE environment variable.

Would it be complicated to add this feature?

**Comment by zandbelt** [comment]
Assuming libcurl provides support for that, it should be fairly easy.

**Reply by zandbelt** [reply]
as it seems from this:
https://everything.curl.dev/usingcurl/tls/sslkeylogfile
adding it as an environment variable in the process that is calling/running Apache would work, perhaps you can confirm?

**Reply by studersi** [reply]
Yes, this is supported by libcurl: https://curl.se/libcurl/c/libcurl-env.html#SSLKEYLOGFILE.
